openpype/pipeline/workfile/lock_workfile.py

```diff
@@ -0,0 +1,67 @@
+import os
+import json
+from uuid import uuid4
+from openpype.lib.pype_info import get_workstation_info
+
+
+def _read_lock_file(lock_filepath):
+    with open(lock_filepath, "r") as stream:
+        data = json.load(stream)
+    return data
+
+
+def _get_lock_file(filepath):
+    return filepath + ".lock"
+
+
+def is_workfile_locked(filepath):
+    lock_filepath = _get_lock_file(filepath)
+    if not os.path.exists(lock_filepath):
+        return False
+    return True
+
+
+def is_workfile_locked_for_current_process(filepath):
+    if not is_workfile_locked():
+        return False
+
+    lock_filepath = _get_lock_file(filepath)
+    process_id = os.environ["OPENPYPE_PROCESS_ID"]
+    data = _read_lock_file(lock_filepath)
+    return data["process_id"] == process_id
+
+
+def delete_workfile_lock(filepath):
```

Comment:
Should this function allow a `force` argument to even force it if not a file of the current user? Or considered too dangerous?

What happens if you force quit (kill process) maya, wouldn't the `.lock` file remain?

Reply:
This is to delete the lock (not matter who created it). This should give ability to "unlock" it e.g. if maya crashes.

Reply:
Function `remove_workfile_lock` is to remove the lock for current process and `delete_workfile_lock` should be used to delete the lock file no matter how is opened.

Reply:
I tested it with` cmds.quit(force =True)` in maya scene with Openpype, the scripts of removing/deleting lock still works even if  it is forced to be closed.

Reply:
I had call with @moonyuet and we did changes that should add enable/disable with settings and ability of ignoring the lock if is existing (so you can take control even if lock is there).

Like I've wrote this is "feature propose" so there are some dirty hacks. We'll do modifications in this PR only if it cause slowdown, it will cause issues or the feature won't work.